app/blueprints/assessments/activity_trail.py

```diff
@@ -0,0 +1,307 @@
+import inspect
+from dataclasses import asdict
+from dataclasses import dataclass
+from dataclasses import field
+from datetime import datetime
+
+from app.blueprints.services.data_services import get_bulk_accounts_dict
+from app.blueprints.services.models.assessor_task_list import AssessorTaskList
+from app.blueprints.services.models.flag import FlagType
+from app.blueprints.tagging.models.tag import AssociatedTag
+from flask import current_app
+from flask_wtf import FlaskForm
+from wtforms import BooleanField
+from wtforms import StringField
+
+
+@dataclass
+class BaseModel:
+    @classmethod
+    def from_list(cls, data_list: list[dict]):
+        return [
+            cls(
+                **{
+                    k: v
+                    for k, v in d.items()
+                    if k in inspect.signature(cls).parameters
+                    and k != "date_created"
+                },
+                date_created=cls._format_date(d.get("date_created")),
+            )
+            for d in data_list
+        ]
+
+    @staticmethod
+    def _format_date(date_str):
+        if date_str:
+            try:
+                current_app.logger.info("Formatting date")
+                return datetime.fromisoformat(date_str).strftime(
+                    "%Y-%m-%d %H:%M:%S"
+                )
+            except ValueError:
+                return date_str
+        return date_str
+
+
+@dataclass
+class AssociatedTags(AssociatedTag):
+    full_name: str = ""
+    email_address: str = ""
+    highest_role: str = ""
+    email_address: str = ""
+    date_created: str = field(default="", metadata={"name": "created_at"})
```

Comment:
Do we need `date_created`? Can't we just use the `created_at` from the parent class?

https://github.com/communitiesuk/funding-service-design-assessment/blob/52e8a270289a80b43883f766c1220ed253372d28/app/blueprints/tagging/models/tag.py#L51

Reply:
But couldn't we make that function use created_at instead? could make it a property method too if all it needs to do is access `date_created`

Reply:
Yes, that is one way of doing it. I just wanted to handle it within the class and thought it looks cleaner 🤔

Reply:
Ah cool can leave if you think this way is better